controllers/objecttemplate_controller_test.go

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"fmt"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"math/rand/v2"
+	client2 "sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+
+	templatesv1alpha1 "github.com/kluctl/template-controller/api/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func buildTestConfigMap(name string, namespace string, data map[string]string) *unstructured.Unstructured {
+	m := map[string]any{}
+	for k, v := range data {
+		m[k] = v
+	}
+	return &unstructured.Unstructured{
```

Comment:
Why Unstructured? I prefer type-safety.

Reply:
Good point. Using typed objects now.

Reply:
Ok, turns out the reason was simple: ObjectTemplate's template entries work with unstructured objects and there is no way around this. I'll at least use typed objects now and then convert them to unstructured.